drupal8-apache/7.0/usr/local/share/drupal/drupal8_functions.sh

```diff
@@ -0,0 +1,113 @@
+#! /usr/bin/env bash
+
+#####
+# Perform the Drupal 8 build.
+#####
+do_drupal_build() {
+  do_drupal_create_directories
+  do_drupal_permissions
+  do_drupal_legacy_install_script
+}
+
+do_drupal_development_build() {
+  do_drupal_build
+  do_drupal_legacy_development_install_script
+}
+
+#####
+# Perform tasks on container start
+#####
+do_drupal_start() {
+  do_drupal_permissions
+  do_drupal_legacy_install_finalise_script
+}
+
+do_drupal_development_start() {
+  do_drupal_start
+  do_drupal_development_install
+  do_drupal_legacy_development_install_script
+}
+
+#####
+# Create any directories that aren't given by default.
+#####
+do_drupal_create_directories() {
+  mkdir -p "${WEB_DIRECTORY}/sites/default/files"
+}
+
+#####
+# Fix directory permissions.
+#####
+do_drupal_permissions() {
+  if [ "$IS_CHOWN_FORBIDDEN" -ne 0 ]; then
+    # Give the docroot and config directories to the web user.
+    chown -R "${APP_USER}":"${APP_GROUP}" "${WEB_DIRECTORY}"
+
+    # Ensure the config directory is writeable by the group
+    if [ -d /app/config ]; then
+      chmod -R g+w /app/config
+      chown -R "${APP_USER}":"${APP_GROUP}" /app/config
+    fi
+
+    # Ensure the files directory is writable.
+    chmod -R g+w "${WEB_DIRECTORY}/sites/default/files"
+
+    # Setting.php needs to be writable during installation, but Drupal will fix
+    # this later.
+    chmod go+w "${WEB_DIRECTORY}/sites/default/settings.php"
+  fi
+}
+
+#####
+# Support the legacy install_custom.sh scripts.
+#####
+do_drupal_legacy_install_script() {
+  if [ -f /usr/local/share/drupal/install_custom.sh ]; then
+    bash /usr/local/share/drupal/install_custom.sh
+  fi
+}
+
+do_drupal_legacy_development_install_script() {
+  if [ -f /usr/local/share/drupal/development/install_custom.sh ]; then
+    bash /usr/local/share/drupal/development/install_custom.sh
+  fi
+}
+
+
+####
+# Support the legacy install_custom_finalise.sh scripts.
+####
+do_drupal_legacy_install_finalise_script() {
+  if [ -f /usr/local/share/drupal/install_finalise.sh ]; then
+    bash /usr/local/share/drupal/install_finalise.sh
+  fi
+}
+
+do_drupal_legacy_development_install_finalise_script() {
+  if [ -f /usr/local/share/drupal/development/install_finalise.sh ]; then
+    bash /usr/local/share/drupal/development/install_finalise.sh
+  fi
+}
+
+#####
+# Install Drupal if required.
+#####
+do_drupal_development_install() {
+
+  if [ "$INSTALL_DRUPAL" == 'false' ]; then
+    return
+  fi
+
+  # Drop the database if we need to force the install every time.
+  if [ "$FORCE_DATABASE_DROP" == 'true' ]; then
+    echo 'Dropping the Drupal DB if it exists'
+    "drush sql-drop -y -r ${WEB_DIRECTORY}"
```

Comment:
Oh, this isn't being executed, which is why it wasn't installing!

If drush should run as root (quotes only around the variable):
`drush sql-drop -y -r "${WEB_DIRECTORY}"` 

If drush should not run as root (ideal solution):
`as_code_owner "drush sql-drop -y -r ${WEB_DIRECTORY}"` 

If the build user does not have permissions, you can either choose to reset permissions before/after or run with `as_app_user` to run as the web server user, though beware any files created will be owned by the web server user.